Source/Scene/Billboard.js

```diff
@@ -762,6 +806,93 @@ define([
         return this._pickId;
     };
 
+    var scratchRay = new Ray();
+    var scratchPosition = new Cartesian3();
+    var scratchCartographic = new Cartographic();
+
+    Billboard._clampPosition = function(object, mode, projection, markObjectDirty) {
+        var modeChanged = mode !== SceneMode.MORPHING && mode !== object._mode;
+        if (object._newTile !== object._currentTile || object._positionChanged || modeChanged) {
+            if (mode === SceneMode.SCENE3D) {
+                Cartesian3.clone(Cartesian3.ZERO, scratchRay.origin);
+                Cartesian3.normalize(object.position, scratchRay.direction);
+            } else {
+                var ellipsoid = projection.ellipsoid;
+                ellipsoid.cartesianToCartographic(object.position, scratchCartographic);
+                scratchCartographic.height = -1000.0; // TODO: get minimum height of entire terrain set
+                projection.project(scratchCartographic, scratchPosition);
+                Cartesian3.fromElements(scratchPosition.z, scratchPosition.x, scratchPosition.y, scratchPosition);
+                Cartesian3.clone(scratchPosition, scratchRay.origin);
+                Cartesian3.clone(Cartesian3.UNIT_X, scratchRay.direction);
+            }
+
+            var position = object._newTile.data.pick(scratchRay, mode, projection, false, scratchPosition);
```

Comment:
I imagine this is already part of the performance plans, but we must be able to optimize the general-purpose ray intersection given we have the specific case of a ray normal to the ellipsoid surface.  I'm talking about the actual number crunching...this is in addition to the spatial data structure / adjacency info / temporal coherence / etc.